sap-vcap-services/src/main/java/com/sap/cloud/environment/servicebinding/SapVcapServicesServiceBindingAccessor.java

```diff
@@ -101,45 +105,72 @@ public List<ServiceBinding> getServiceBindings()
         List<ServiceBinding>
         extractServiceBindings( @Nonnull final JSONObject vcapServices, @Nonnull final String serviceName )
     {
-        final JSONArray jsonServiceBindings;
-        try {
-            jsonServiceBindings = vcapServices.getJSONArray(serviceName);
-        }
-        catch( final JSONException e ) {
-            logger.debug("Skipping '{}': Unexpected format.", VCAP_SERVICES);
+        final Object serviceBindings = vcapServices.get(serviceName);
+        if( !(serviceBindings instanceof JSONArray) ) {
+            logger.debug("Skipping '{}': Unexpected format.", serviceName);
             return Collections.emptyList();
         }
 
-        final List<ServiceBinding> serviceBindings = new ArrayList<>(jsonServiceBindings.length());
-        for( int i = 0; i < jsonServiceBindings.length(); ++i ) {
-            final JSONObject jsonServiceBinding;
-            try {
-                jsonServiceBinding = jsonServiceBindings.getJSONObject(i);
-            }
-            catch( final JSONException e ) {
-                continue;
-            }
-
-            serviceBindings.add(toServiceBinding(jsonServiceBinding, serviceName));
+        final List<ServiceBinding> result =
+            StreamSupport
+                .stream(((Iterable<?>) serviceBindings).spliterator(), false)
+                .filter(JSONObject.class::isInstance)
+                .flatMap(binding -> toServiceBindings(((JSONObject) binding).toMap(), serviceName).stream())
+                .collect(Collectors.toList());
+
+        logger.debug("Successfully read {} service binding(s) from '{}'.", result.size(), serviceName);
+        return result;
+    }
+
+    @Nonnull
+    protected
```

Comment:
I'm always a bit hesitant with introducing new public API in this library.
Do you have strong feelings about these "extensibility points" or do you think delaying this (making the methods `protected`) until it is actually requested is also fine?

Reply:
That'd be fine too. Maybe I am just not 100% confident that our assumptions hold true, for us and our users.